espnet2/asr_transducer/decoder/stateless_decoder.py

```diff
@@ -0,0 +1,145 @@
+"""Stateless decoder definition for Transducer models."""
+
+from typing import List, Optional, Tuple, Union
+
+import torch
+from typeguard import check_argument_types
+
+from espnet2.asr_transducer.beam_search_transducer import ExtendedHypothesis, Hypothesis
+from espnet2.asr_transducer.decoder.abs_decoder import AbsDecoder
+
+
+class StatelessDecoder(AbsDecoder):
+    """Stateless Transducer decoder module.
+
+    Args:
+        vocab_size: Output size.
+        embed_size: Embedding size.
+        embed_dropout_rate: Dropout rate for embedding layer.
+        embed_pad: Embed/Blank symbol ID.
+
+    """
+
+    def __init__(
+        self,
+        vocab_size: int,
+        embed_size: int = 256,
+        embed_dropout_rate: float = 0.0,
+        embed_pad: int = 0,
+    ) -> None:
+        assert check_argument_types()
+
+        super().__init__()
+
+        self.embed = torch.nn.Embedding(vocab_size, embed_size, padding_idx=embed_pad)
+        self.embed_dropout_rate = torch.nn.Dropout(p=embed_dropout_rate)
+
+        self.output_size = embed_size
+        self.vocab_size = vocab_size
+
+        self.blank_id = embed_pad
+
+        self.device = next(self.parameters()).device
+        self.score_cache = {}
+
+    def forward(
+        self,
+        labels: torch.Tensor,
+        states: Optional[Tuple[torch.Tensor, Optional[torch.Tensor]]] = None,
+    ) -> Tuple[torch.Tensor, None]:
```

Comment:
The type hint for the return value does not match the actual return value and hence it is misleading.

Reply:
What a timing, I was fixing a bunch of docstrings and type hints! Sorry for that!
Thanks!